pkgs/test_api/lib/src/backend/compiler.dart

```diff
@@ -41,8 +41,8 @@ enum Compiler {
   /// [Compiler].
   ///
   /// Note that custom [Compiler] implementations are not supported.
-  factory Compiler.deserialize(Object serialized) => builtIn
-      .firstWhere((compiler) => compiler.identifier == serialized as String);
+  factory Compiler.deserialize(String serialized) =>
```

Comment:
This is technically breaking, this is a public API (part of `backend.dart`).

Reply:
But there is NO world where this would run without failing, right?

Reply:
I am not totally sure... but probably this change is OK in practice?

Reply:
since we're casting `as String`?

Reply:
I think this is OK. This constructor is not used outside of the package and I don't think it's worth a major version bump since we only support our known uses.